### PR TITLE
fmt: Add fmt_strnlen function

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

Add an `fmt_strnlen` function, equivalent to the POSIX `strnlen` function. PR includes a test for the new function.

### Issues/PRs references

None

### CODE REVIEW

File: sys/fmt/fmt.c [hunk 1]
```diff
@@ -83,6 +83,16 @@ size_t fmt_strlen(const char *str)
     return (tmp - str);
 }
 
+size_t fmt_strnlen(const char *str, size_t maxlen)
+{
+    const char *tmp = str;
+    const char *end = str + maxlen;
```
Comment (kaspar030) [inline sys/fmt/fmt.c]:
```
size_t fmt_strnlen(const char *str, size_t maxlen)
{
    const char *tmp = str;
    while(*tmp && maxlen--) {
        tmp++;
    }
    return (tmp - str);
}
```

... would more fit the style of the rest of the file.

Comment (bergzand) [reply sys/fmt/fmt.c]: Agreed, fixed

**Comment by kaspar030** [comment]
please squash!

**Comment by bergzand** [comment]
squashed!

**Comment by bergzand** [comment]
@kaspar030 Murdock is failing on unrelated tests runs on the samr21-xpro

**Comment by cgundogan** [comment]
@bergzand looks like a murdock hickup. I triggered a rebuild and that made murdock great again :+1:

**Pull request merged**